src/cargo/core/shell.rs

```diff
@@ -76,12 +95,34 @@ impl MultiShell {
         self.err().say(message, YELLOW)
     }
 
+    pub fn set_verbosity(&mut self, verbose: bool, quiet: bool) {
+        self.verbosity = match (verbose, quiet) {
+            (true, true) => {
+                panic!("You cannot set both --verbose and --quiet");
```

Comment:
Cargo in general tries to not panic as much as possible, can this function return a `CargoResult` which is then handled through the normal means?